test/common/grpc/codec_test.cc

```diff
@@ -106,6 +107,41 @@ TEST(GrpcCodecTest, decodeInvalidFrame) {
   EXPECT_EQ(size, buffer.length());
 }
 
+// A frame of nothing followed by a frame of 0's followed then invalid
+TEST(GrpcCodecTest, decodeNullByteFrameWithInvalidFrameAfterward) {
+  const std::string data("\000\000\000\000\0000000", 9);
+  Buffer::OwnedImpl buffer(data.data(), data.size());
+  std::vector<Frame> frames;
+  Decoder decoder;
+  ENVOY_LOG_MISC(trace, "Buffer: {}", buffer.toString());
+  EXPECT_FALSE(decoder.decode(buffer, frames));
+  // When the decoder doesn't successfully decode, it puts valid frames up until
+  // an invalid frame into output frame vector.
+  EXPECT_NE(0, frames.size());
+}
+
+// If there is valid frames followed by an invalid frame, the decoder will succesfully put the valid
+// frame in the output and return false due to the invalid frame
+TEST(GrpcCodecTest, decodeValidFramesWithInvalidFrameAfterward) {
+  // Decode a valid encoded structured response plus invalid data afterward
+  grpc::health::v1::HealthCheckResponse response;
```

Comment:
(just FYI, you could also have used the `helloworld::HelloRequest` frame message to make this test abstracted from health checking)

Reply:
I'll change it!